docs/concepts/middleware.md

```diff
@@ -90,3 +90,29 @@ export const handler = [
   },
 ];
 ```
+
+## Middleware Destination
+
+See these [two](https://github.com/denoland/fresh/issues/521)
+[issues](https://github.com/denoland/fresh/issues/1341) which led to this
+[PR](https://github.com/denoland/fresh/pull/1123), but there is now a way to
```

Comment:
I feel like this section should provide a small blurb about what you can do with middleware destinations instead of pointing at issues in the tracker. What do you think?

Reply:
I tried to communicate this with the following section:
```
This is useful for if you want your middleware to only run when a request is
headed for a `route`, as opposed to something like
`http://localhost:8001/favicon.ico`.
```
Perhaps I should expand upon this? And maybe I should put this at the top and put the issues and PR at the bottom, for reference.

Reply:
I've completely redone it. Thanks for holding a high bar of quality, @marvinhagemeister.

Perhaps we need a followup example that actually has useful code (as opposed to what I have here, which is just logging values to the console). If you want, I can add the example here. Something like "Running middleware only for routes". Thoughts?